### PR TITLE
feat(filter): Add less/sass/test page for filter component

### PR DESCRIPTION
**Pull request by jeff-phillips-18**
## Description
Adds necessary selectors for filter components.

## Link to rawgit and/or image
http://rawgit.com/jeff-phillips-18/patternfly/filter-dist/dist/tests/filter.html

## PR checklist (if relevant)

- [ ] **Cross browser**: works in IE9
- [ ] **Cross browser**: works in IE10
- [ ] **Cross browser**: works in IE11
- [ ] **Cross browser**: works in Edge
- [x] **Cross browser**: works in Chrome
- [x] **Cross browser**: works in Firefox
- [x] **Cross browser**: works in Safari
- [ ] **Cross browser**: works in Opera
- [x] **Responsive**: works in extra small, small, medium and large view ports.
- [x] **Preview the PR**: An image or a URL for designer to preview this PR is provided.

### CODE REVIEW

File: src/less/filter.less [hunk 1]
```diff
@@ -0,0 +1,44 @@
+.filter-pf {
+  .filter-pf-category-select {
```
Comment (andresgalante) [inline src/less/filter.less]: Let's avoid nesting this selectors, it'll just increase specificty

**Comment by cliffpyles** [comment]
Other than the nested selector mentioned above, it looks good to me. 👍

File: src/less/filter.less (hunk 1 above)
Comment (jeff-phillips-18) [reply src/less/filter.less]: Fixed.

**Review by matthewcarleton — DISMISSED** [review]
👍 Although the screen shot doesn't match with what I'm seeing in the browser. The category filters are stacked. 
screen shot 2018-01-03 at 1 12 25 pm
Are we concerned with that?

**Comment by jeff-phillips-18** [comment]
Sorry, my dist branch was built incorrectly. Should be accurate now.

File: src/less/filter.less [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+.filter-pf-category-item {
+  margin-bottom: 5px;
+}
+.filter-pf-category-label {
+  font-weight: 700;
+  margin-right: 5px;
+  padding: 5px 0 6px 5px;
+}
+.filter-pf-fields .form-group {
+  width: 275px;
```
Comment (andresgalante) [inline src/less/filter.less]: The width of the component should be regulated by it's parent container, I don't think we should not define a fix width.

File: src/less/filter.less [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+}
+.filter-pf-select {
+  .dropdown.btn-group {
+    width: 100%;
```
Comment (andresgalante) [inline src/less/filter.less]: Are you sure we want to extend this element to the width of it's container? honestly I don't know :)

File: src/less/filter.less [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  }
+  .dropdown-menu {
+    min-width: 100%;
+  }
+  .caret {
+    margin-top: -5px;
+    position: absolute;
+    top: 50%;
+    right: 12px;
```
Comment (andresgalante) [inline src/less/filter.less]: this is probably 10 or 15, in general in pf3 we deal with multiples of 5

File: src/less/filter.less [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+}
+.filter-pf-select {
+  .dropdown.btn-group {
+    width: 100%;
+  }
+  .dropdown-menu {
+    min-width: 100%;
+  }
+  .caret {
+    margin-top: -5px;
```
Comment (andresgalante) [inline src/less/filter.less]: the hack for vertical centering is not a negative margin top but a `transform: translateY(-50%);`

File: src/less/filter.less [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    top: 50%;
+    right: 12px;
+  }
+}
+.filter-pf-select-dropdown {
+  background-color: @color-pf-white;
+  background-image: none;
+  color: @color-pf-black-500;
+  font-size: 12px;
+  font-style: italic;
```
Comment (andresgalante) [inline src/less/filter.less:28]: This `italic` is making the caret look weird.

File: src/less/filter.less [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    min-width: 100%;
+  }
+  .caret {
+    margin-top: -5px;
+    position: absolute;
+    top: 50%;
+    right: 12px;
+  }
+}
+.filter-pf-select-dropdown {
```
Comment (andresgalante) [inline src/less/filter.less]: I am ok with all this font treatment as long as we make sure that it's consistent with other placeholders for inputs.

File: src/sass/converted/patternfly/_filter.scss [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+  color: $color-pf-black-500;
+  font-size: 12px;
+  font-style: italic;
+  font-weight: 400;
+  padding-right: 25px;
+  text-align: left;
+  width: 100%;
+}
+
+.filter-pf-active-label {
```
Comment (andresgalante) [inline src/sass/converted/patternfly/_filter.scss]: Where is this label in the HTML?

File: src/less/filter.less [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+  background-color: @color-pf-white;
+  background-image: none;
+  color: @color-pf-black-500;
+  font-size: 12px;
+  font-style: italic;
+  font-weight: 400;
+  padding-right: 25px;
+  text-align: left;
+  width: 100%;
```
Comment (andresgalante) [inline src/less/filter.less]: if you remove the other 100%, maybe remove this one too

**Comment by andresgalante** [comment]
@jeff-phillips-18 I left you a bunch of comments, I hope they make sense 😄

File: src/sass/converted/patternfly/_filter.scss (hunk 8 above)
Comment (jeff-phillips-18) [reply src/sass/converted/patternfly/_filter.scss]: This is another item on the toolbar showing current filters, this is not shown in the test page

File: src/less/filter.less (hunk 3 above)
Comment (jeff-phillips-18) [reply src/less/filter.less]: You are correct, these are not necessary.

File: src/less/filter.less (hunk 4 above)
Comment (jeff-phillips-18) [reply src/less/filter.less]: 10px seems to work fine.

File: src/less/filter.less (hunk 9 above)
Comment (jeff-phillips-18) [reply src/less/filter.less]: done

**Comment by jeff-phillips-18** [comment]
Updated per comments. Thanks @andresgalante

**Comment by jeff-phillips-18** [comment]
Regression test failures do not happen locally, only in travis.

```
$ npm run regressions pagination-table-view

> patternfly@0.0.0-semantically-released regressions /Users/jeffreyphillips/repositories/patternfly
> node ./backstop/test "pagination-table-view"

Config detected:
COMMAND | Executing core for `test`
createBitmaps | Selected 1 of 1 scenarios.
Starting Chromy: {"chromeFlags":["--disable-gpu","--force-device-scale-factor=1","--disable-infobars=true","--window-size=1280,720"],"port":9222,"waitTimeout":30000,"visible":false}
9222 Chrome v63 detected.
9222 INFO >  BackstopTools have been installed.
      COMMAND | Executing core for `report`
      compare | OK: pagination-table-view backstop_default_pagination-table-view_0_document_0_large-device.png
       report | Test completed...
       report | 1 Passed
       report | 0 Failed
       report | Writing browser report
       report | Browser reported copied
       report | Copied configuration to: /Users/jeffreyphillips/repositories/patternfly/backstop_data/html_report/config.js
      COMMAND | Executing core for `openReport`
   openReport | Opening report.
      COMMAND | Command `report` successfully executed in [0.762s]
      COMMAND | Command `test` successfully executed in [4.624s]
tests completed without failures



$ npm run regressions table-view-navbar

> patternfly@0.0.0-semantically-released regressions /Users/jeffreyphillips/repositories/patternfly
> node ./backstop/test "table-view-navbar"

Config detected:
COMMAND | Executing core for `test`
createBitmaps | Selected 1 of 1 scenarios.
Starting Chromy: {"chromeFlags":["--disable-gpu","--force-device-scale-factor=1","--disable-infobars=true","--window-size=1280,720"],"port":9222,"waitTimeout":30000,"visible":false}
9222 Chrome v63 detected.
9222 INFO >  BackstopTools have been installed.
      COMMAND | Executing core for `report`
      compare | OK: table-view-navbar backstop_default_table-view-navbar_0_document_0_large-device.png
       report | Test completed...
       report | 1 Passed
       report | 0 Failed
       report | Writing browser report
       report | Browser reported copied
       report | Copied configuration to: /Users/jeffreyphillips/repositories/patternfly/backstop_data/html_report/config.js
      COMMAND | Executing core for `openReport`
   openReport | Opening report.
      COMMAND | Command `report` successfully executed in [0.467s]
      COMMAND | Command `test` successfully executed in [3.812s]
tests completed without failures
```

**Comment by jeff-phillips-18** [comment]
The test should not have worked locally. I have fixed and updated the changed files. Changes were do to removing the 25% width on filter components in a toolbar.

**Review by andresgalante — APPROVED** [review]

**Pull request merged**